template/src/services/networking/axios.js

```diff
@@ -0,0 +1,79 @@
+import axios from 'axios'
+import {setToken, TokenType} from '.'
+import {AXIOS_TIMEOUT, REFRESH_TOKEN_KEY, TOKEN_KEY, UNAUTHORIZED} from '../../constants'
+import {getRefreshToken} from '../api'
+import {clearAllKeys, getString, setData} from '../../services/mmkv/storage'
+import RouteKey from '../../navigation/RouteKey'
+import Config from 'react-native-config'
+import {store} from '../../store/store'
+import {appActions} from '../../store/reducers'
+
+const instance = axios.create({
+  baseURL: Config.API_URL,
+  timeout: AXIOS_TIMEOUT,
+  withCredentials: false,
+  responseType: 'json',
+  headers: {
+    Accept: 'application/json',
+    'Content-Type': 'application/json; charset=utf-8',
+  },
+  data: {},
+})
+
+const backToLogin = () => {
+  store.dispatch(appActions.setAppStack(RouteKey.AuthStack))
+}
+
+const handleRefreshToken = async (refreshToken, originalConfig) => {
+  // Call RefreshToken API
+  return getRefreshToken(refreshToken)
+    .then(response => {
+      // Save new Token and RefreshToken
+      setToken(response?.data?.token, TokenType.Bearer)
+      setData(TOKEN_KEY, response?.data?.token)
+      setData(REFRESH_TOKEN_KEY, response?.data?.refreshToken)
+      return instance(originalConfig)
+    })
+    .catch(() => {
+      // Remove all keys and back to login screen to get new token
+      clearAllKeys()
+      backToLogin()
```

Comment:
We should call logout action. because we need to clear data in redux as well

Reply:
done